packages/opencensus-core/src/metrics/cumulative/cumulative.ts

```diff
@@ -0,0 +1,196 @@
+   * labels.
+   *
+   * @returns {CumulativePoint} The value of single cumulative.
+   */
+  getDefaultTimeSeries(): CumulativePoint {
+    return this.registerTimeSeries(this.defaultLabelValues);
+  }
+
+  /**
+   * Removes the TimeSeries from the cumulative metric, if it is present. i.e.
+   * references to previous Point objects are invalid (not part of the
+   * metric).
+   *
+   * @param {LabelValue[]} labelValues The list of label values.
+   */
+  removeTimeSeries(labelValues: LabelValue[]): void {
+    validateNotNull(labelValues, 'labelValues');
+    this.registeredPoints.delete(hashLabelValues(labelValues));
+  }
+
+  /**
+   * Removes all TimeSeries from the cumulative metric. i.e. references to all
+   * previous Point objects are invalid (not part of the metric).
+   */
+  clear(): void {
+    this.registeredPoints.clear();
+  }
+
+  /**
+   * Registers a TimeSeries and returns a Point if the specified
+   * labelValues is not already associated with this cumulative, else returns an
+   * existing Point.
+   *
+   * @param {LabelValue[]} labelValues The list of the label values.
+   * @returns {CumulativePoint} The value of single cumulative.
+   */
+  private registerTimeSeries(labelValues: LabelValue[]): CumulativePoint {
+    const hash = hashLabelValues(labelValues);
+    // return if the specified labelValues is already associated with the point.
+    if (this.registeredPoints.has(hash)) {
+      return this.registeredPoints.get(hash)!;
+    }
+    if (this.labelKeysLength !== labelValues.length) {
+      throw new Error('Label Keys and Label Values don\'t have same size');
+    }
+
+    const point =
+        new CumulativePointEntry([...labelValues, ...this.constantLabelValues]);
+    this.registeredPoints.set(hash, point);
+    return point;
+  }
+
+  /**
+   * Provides a Metric with one or more TimeSeries.
+   *
+   * @returns {Metric} The Metric.
+   */
+  getMetric(): Metric|null {
+    if (this.registeredPoints.size === 0) {
+      return null;
+    }
+    const now: Timestamp = getTimestampWithProcessHRTime();
+    return {
+      descriptor: this.metricDescriptor,
+      timeseries: Array.from(
+          this.registeredPoints, ([_, point]) => point.getTimeSeries(now))
+    };
+  }
+}
+
+/**
+ * The value of a single point in the Cumulative.TimeSeries.
+ */
+export class CumulativePointEntry implements CumulativePoint {
+  private readonly labelValues: LabelValue[];
+  private startTimestamp: Timestamp;
+  private value = 0;
+
+  constructor(labelValues: LabelValue[]) {
+    this.labelValues = labelValues;
+    this.startTimestamp = getTimestampWithProcessHRTime();
+  }
+
+  /** Reset cumulative metric. */
+  reset(): void {
+    this.value = 0;
+    this.startTimestamp = getTimestampWithProcessHRTime();
+  }
+
+  /**
+   * Increment the cumulative metric.
+   * @param {number} val The new value.
+   */
+  inc(val?: number): void {
+    if (val && !Number.isFinite(val)) {
```

Comment:
Does this catch `NaN` (I'm guessing not), and assuming not, could you make it do that?